compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5228,6 +5228,11 @@ private BType checkInvocationArgs(BLangInvocation iExpr, List<BType> paramTypes,
             requiredParams.add(nonRestParam);
         }
 
+        // check argument types in arr:sort function
+        if (iExpr.langLibInvocation && iExpr.name.value.equals("sort")) {
```

Comment:
In org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker#checkInLangLib we do some langlib specific validations. Wondering if we should move this also there.